destination.go

```diff
@@ -166,6 +166,7 @@ func (a *destinationPluginAdapter) Open(ctx context.Context, _ pconnector.Destin
 
 func (a *destinationPluginAdapter) Run(ctx context.Context, stream pconnector.DestinationRunStream) error {
 	ctx = internal.Enrich(ctx, a.cfg)
+	a.writeStrategy.SetStream(stream.Server())
 
 	for stream := stream.Server(); ; {
```

Comment:
That line overwrites the `stream` variable with a value of a different type. It's a bit of a trick, you can do the same in switch statements for instance ([example](https://github.com/ConduitIO/conduit-commons/blob/f04a489de6d588c2dabfa7f9b7f1b2f1354e7b27/opencdc/record.go#L104)). We use it for convenience (no need to figure out a different name for the server stream) but also to prevent us from using the `stream` variable that's active outside of the loop, since we have no use for it anymore.

Reply:
Sweet, thanks for the explanation!